Doc/whatsnew/3.12.rst

```diff
@@ -62,91 +62,171 @@ Summary -- Release highlights
+to the language and the standard library.
+The library changes focus on cleaning up deprecated APIs, usability, and correctness.
+Of note, the :mod:`!distutils` package has been removed from the standard library.
+Filesystem support in :mod:`os` and :mod:`pathlib` has seen a number of improvements,
+and several modules have better performance.
+
+The language changes focus on usability,
+as :term:`f-strings <f-string>` have had many limitations removed
+and 'Did you mean ...' suggestions continue to improve.
+The new :ref:`type parameter syntax <whatsnew312-pep695>`
+and :keyword:`type` statement improve ergonomics for using :term:`generic types
+<generic type>` and :term:`type aliases <type alias>` with static type checkers.
+
+This article doesn't attempt to provide a complete specification of all new features,
+but instead tries to give a convenient overview.
+For full details, you should refer to the documentation,
+such as the  :ref:`Library Reference <library-index>`
+and :ref:`Language Reference <reference-index>`.
+If you want to understand the complete implementation and design rationale for a change,
+refer to the PEP for a particular new feature;
+but note that PEPs usually are not kept up-to-date
+once a feature has been fully implemented.
+
+--------------
 
 .. PEP-sized items next.
 
+New syntax features:
+
+* :ref:`PEP 695 <whatsnew312-pep695>`, type parameter syntax and the :keyword:`type` statement
+
 New grammar features:
 
-* :ref:`whatsnew312-pep701`
+* :ref:`PEP 701 <whatsnew312-pep701>`, :term:`f-strings <f-string>` in the grammar
 
 Interpreter improvements:
 
-* :ref:`whatsnew312-pep684`
+* :ref:`PEP 684 <whatsnew312-pep684>`, a unique per-interpreter :term:`GIL
+  <global interpreter lock>`
+* :ref:`PEP 669 <whatsnew312-pep669>`, low impact monitoring
+* `Improved 'Did you mean ...' suggestions <improved error messages_>`_
+  for :exc:`NameError`, :exc:`ImportError`, and :exc:`SyntaxError` exceptions
 
-* :ref:`whatsnew312-pep669`
+Python data model improvements:
 
-New typing features:
+* :ref:`PEP 688 <whatsnew312-pep688>`, using the :ref:`buffer protocol
+  <bufferobjects>` from Python
+
+Significant improvements in the standard library:
+
+* The :class:`pathlib.Path` class now supports subclassing
+* The :mod:`os` module received several improvements for Windows support
+* A :ref:`command-line interface <sqlite3-cli>` has been added to the
+  :mod:`sqlite3` module
+* :func:`isinstance` checks against :func:`runtime-checkable protocols
+  <typing.runtime_checkable>` enjoy a speed up of between two and 20 times
+* A :ref:`command-line interface <uuid-cli>` has been added to the
+  :mod:`uuid` module
 
-* :ref:`whatsnew312-pep688`
+Security improvements:
 
-* :ref:`whatsnew312-pep692`
+* Replace the builtin :mod:`hashlib` implementations of
+  SHA1, SHA3, SHA2-384, SHA2-512, and MD5 with formally verified code from the
+  `HACL* <https://github.com/hacl-star/hacl-star/>`__ project.
```

Comment:
I went through the changelog for security items, it seemed HACL* was worth noting. I'd welcome suggestions for any other security changes in 3.12 we should note.

cc: @gpshead

Reply:
This makes sense to mention, though it is worth pointing out that when linked against OpenSSL, which we do in our binary releases and most Linux distros do, the OpenSSL implementations of the algorithms are used first when available.